### ISSUE TITLE
A serious Blind Spot in the opening joseki

### ISSUE THREAD

**Issue by poptangtwe** [lightvector/KataGo#989]
![1-1](https://github.com/user-attachments/assets/474a74a5-32f0-427b-bb22-74c30cb3f888)
![1-2](https://github.com/user-attachments/assets/4e0d6703-4664-490e-926b-f61b1f7be292)
![1-3](https://github.com/user-attachments/assets/9f7d40e8-0c90-4bd3-8683-9ffe6501627f)
![1-4](https://github.com/user-attachments/assets/002f95f9-35e0-425e-9d16-9b93a404f75a)
![2-1](https://github.com/user-attachments/assets/101c99c0-622a-44bb-8e07-e98f6da8f2ab)
![2-2](https://github.com/user-attachments/assets/5153e952-9ad2-499a-a0c2-93eeeb4ed830)
![2-3](https://github.com/user-attachments/assets/61721cb1-f7c8-4ecc-986f-83d7dcd25faf)
![2-4](https://github.com/user-attachments/assets/2a5cd4f2-efbb-44b0-9056-1ddecc34f9d6)
![3-1](https://github.com/user-attachments/assets/062421df-d067-4574-93d4-d55b517f37c6)
![3-2](https://github.com/user-attachments/assets/4b8c1b56-fe94-4f18-b428-993662ee55b5)

**Comment by lightvector** [comment]
Thanks for posting this. I will add more variations of this kind of pincer to the positions for self-play next time I update the positions, likely within the next month or so, then most likely it should gradually sharpen and improve on those variations over the next months after that.

If you know of more positions that have significant winrate swings like this in common opening joseki, post them as well and I can add variations for those too. I suspect that improving on these kinds of patterns are among the more influential of misevaluations to fix because many blind spots that are specific to a deep midgame fight may almost never occur in another game again, but joseki errors have the potential to happen over and over across many games.

**Comment by Arce0416** [comment]
Hi，lightvector，This method actually works! So, can I apply it to KataGo for Gomoku (with 26 opening moves in Gomoku)? How effective would this approach generally be in KataGo for Go?

**Comment by Arce0416** [comment]
> 感谢您发布此信息。下次更新位置时，我将在位置中添加更多此类钳形变化以供自玩，可能在下个月左右，然后很可能在接下来的几个月内逐渐完善和改进这些变化。
> 
> 如果您知道更多常见的开局定式中胜率波动幅度如此大的局面，请发布它们，我也可以添加这些变化。我怀疑改进这些类型的模式是需要纠正的更具影响力的错误评估之一，因为许多特定于中局深处战斗的盲点可能几乎永远不会再出现在另一场比赛中，但定式错误有可能在许多比赛中反复发生。

  Hi，lightvector，This method actually works? So, can I apply it to KataGo for Gomoku (with the 26 standard opening moves)? How effective would it generally be in KataGo for Go?

  I once gave similar advice to sigmoid(hzy) (suggesting adding variations to certain difficult opening blind spots during self-play), but it was rejected because the effectiveness was unknown and might not be useful (or significant). So, I want to know how much impact this would have on KataGo for Go.

**Comment by Arce0416** [comment]
> Thanks for posting this. I will add more variations of this kind of pincer to the positions for self-play next time I update the positions, likely within the next month or so, then most likely it should gradually sharpen and improve on those variations over the next months after that.
> 
> If you know of more positions that have significant winrate swings like this in common opening joseki, post them as well and I can add variations for those too. I suspect that improving on these kinds of patterns are among the more influential of misevaluations to fix because many blind spots that are specific to a deep midgame fight may almost never occur in another game again, but joseki errors have the potential to happen over and over across many games.

Hi，lightvector，This method actually works? So, can I apply it to KataGo for Gomoku (with the 26 standard opening moves)? How effective would it generally be in KataGo for Go?

I once gave similar advice to sigmoid(hzy) (suggesting adding variations to certain difficult opening blind spots during self-play), but it was rejected because the effectiveness was unknown and might not be useful (or significant). So, I want to know how much impact this would have on KataGo for Go.

**Comment by lightvector** [comment]
@Arce0416 yes, this method does work so long as the shape that contains the mistake is one that is likely to reoccur. In general, if you include data about all the different variations of a position that was misevaluated, the training process can learn to evaluate that position correctly. The main challenge is that for an arbitrary mistake, the exact shape that led to that mistake may be unlikely to occur again in any future game because there are exponentially many possible shapes, so the effort to fix that mistake may not be worth much. However, if it is a mistake in a repeatable variation or a kind of mistake that you observe happening multiple times across many games, then there can be a clear benefit to fix that mistake.

**Comment by lightvector** [comment]
This was how KataGo learned to evaluate Mi Yuting's "flying dagger" joseki more accurately - this joseki is a widespread weakness of pure "zero" AI bots because it's sharp enough that it requires specific training but AlphaZero-based methods tend not to explore it well. Also this is how KataGo learned to evaluate certain difficult 3-4 pincer joseki better. Learning these positions through manual intervention to add more variations from these josekis to starting play was definitely worth it, because these positions continue to occur regularly in future games, so unlike fixing a mistake that might never happen again in any future game, fixing these mistakes definitely has an ongoing benefit for improving analysis in many future games.

**Comment by poptangtwe** [comment]
![image](https://github.com/user-attachments/assets/42b9d44b-9d56-4c70-8ba2-a1367ce6e9ee)
https://katagotraining.org/sgfplayer/training-games/65461976/
28B_s797 still does not correctly handle this opening joseki.

**Comment by lightvector** [comment]
So it would be much too soon to see anything, maybe check back in a couple of months, now that the hint position is there?

**Comment by poptangtwe** [comment]
28B_s826 can find the correct move in this kind of pincer joseki. Thanks a lot.❤️@lightvector 
https://katagotraining.org/sgfplayer/training-games/67175665/
![Image](https://github.com/user-attachments/assets/88d041d1-5570-4dec-9c46-ed5ff480409f)
